vscode-trace-extension/src/extension.ts

```diff
@@ -69,12 +69,25 @@ export function activate(context: vscode.ExtensionContext): ExternalAPI {
         })
     );
 
+    const updateUris = async (): Promise<void> => {
+         // If the extension is running remotely, this function automatically establishes a port forwarding tunnel
+         //  from the local machine to `target` on the remote and returns a local uri to the tunnel.
+        const baseUri = vscode.Uri.parse(getTraceServerUrl());
+        const extUri = await vscode.env.asExternalUri(baseUri);
+        const extUriString = extUri.toString();
+
+        tracesProvider.updateTraceServerUrl(extUriString);
+        myAnalysisProvider.updateTraceServerUrl(extUriString);
+        TraceViewerPanel.updateTraceServerUrl(extUriString);
+    };
+
     const analysisProvider = new AnalysisProvider();
     // TODO: For now, a different command opens traces from file explorer. Remove when we have a proper trace finder
     const fileOpenHandler = fileHandler(analysisProvider);
     context.subscriptions.push(
         vscode.commands.registerCommand('traces.openTraceFile', async (file: vscode.Uri) => {
             await startTraceServerIfAvailable(file.fsPath);
+            await updateUris();
```

Comment:
I don't think we should call it at every open. updateUris() will each webview (e.g. open traces) with the new url and the tspclient instance is refreshed. This will require a refressh of each webview which doesn't exist. I think the URLs should be only updated when changed.